Stitch/Graph/StitchAI/Mapping/SyntaxToActions/deriveLayer.swift

```diff
@@ -964,6 +985,138 @@ extension SyntaxViewName {
     }
 }
 
+func parseMathExpressionToPatchNodes(
+    _ mathSyntax: SyntaxViewMathSyntax,
+    varName: String?,
+    viewEvent: SyntaxViewEvent?,
+    nodesDict: [UUID: NodeEntity],
+    isStreaming: Bool
+) throws -> [PatchSyntaxResultType] {
+
+    // Recursively get patch data for operands
+    let lhsResults = try SyntaxViewName.derivePortValues(
+        from: mathSyntax.lhs,
+        varName: varName,
+        viewEvent: viewEvent,
+        nodesDict: nodesDict,
+        isStreaming: isStreaming
+    )
+
+    let rhsResults = try SyntaxViewName.derivePortValues(
+        from: mathSyntax.rhs,
+        varName: varName,
+        viewEvent: viewEvent,
+        nodesDict: nodesDict,
+        isStreaming: isStreaming
+    )
+
+    // Map operator to patch type
+    let patchType = try operatorToPatchType(mathSyntax.op)
+
+    // Create math patch node
+    let mathNodeId = UUID()
```

Comment:
Derive from varName whenever possible:
```suggestion
    let mathNodeId = varName != nil ? deterministicUUID(from: varName!) : UUID()
```